packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/ScatterPlotViewer/helpers/utils.js

```diff
@@ -1,10 +1,47 @@
 import { scaleLinear } from '@vx/scale'
 import * as d3 from 'd3'
+import { flatten, zipWith } from 'lodash'
+
+function isThisMultipleDataSeries (data) {
+  return Array.isArray(data)
+}
+
+export function getDataPoints (data) {
+  if (isThisMultipleDataSeries(data)) {
+    return data
+  } else {
+    return [
+      { seriesData: zipWith(data.x, data.y,
+        function (a, b) {
+          return { x: a, y: b }
+        }) 
+      }
+    ]
+  }
+}
 
 export function getDataExtent (data) {
-  return {
-    x: d3.extent(data.x),
-    y: d3.extent(data.y)
+  if (isThisMultipleDataSeries(data)) {
+    const xValues = flatten(data.map((series) => {
```

Comment:
This might not be the most performant option, since it's iterating over the same data twice (once for `x`, once for `y`) for each series. It may be worth profiling with real data before making any changes though.

Reply:
Yes, I'd prefer to profile with an example of real data. I just don't really know quite yet what that looks like.